sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/job_definition.py

```diff
@@ -0,0 +1,52 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Dict, Union, Any
+
+from azure.ai.ml._schema._deployment.batch.job_definition_schema import JobDefinitionSchema
+from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY
+from azure.ai.ml.entities._component.component import Component
+
+
+class JobDefinition:
+    """Job Definition entity
+
+    :param job: Job definition
+    :type job: Union[Component, str]
+    :param component: Component definition
+    :type component: Union[Component, str]
+    :param type: Job definition type
+    :type type: str
```

Comment:
Job def type is an arbitrary str? We don't have an enum to use?

Reply:
Yes, we validate that this str is valid in job_definition schema. For now, the only valid value is `pipeline`